### PR TITLE
[zh-cn] sync blog:2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md

### PR DESCRIPTION
**Pull request by my-git9**

### CODE REVIEW

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+-->
+## 如何解决此问题？
+
+{{< figure src="/images/blog/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades/mvp-flow-diagram.svg" class="diagram-large" >}}
+
+<!--
+The new feature “Mixed Version Proxy” provides the kube-apiserver with the capability to proxy a request to a peer kube-apiserver which is aware of the requested resource and hence can serve the request. To do this, a new filter has been added to the handler chain in the API server's aggregation layer.
+-->
+"混合版本代理"特性为 kube-apiserver 提供了将请求代理到对等 kube-apiserver 的能力，
+该对等 kube-apiserver 知道所请求的资源，因此可以为此请求提供服务。为此，API
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:64]:
```suggestion
"混合版本代理"新特性为 kube-apiserver 提供了将请求代理到对等 kube-apiserver
以感知请求的资源并因此提供请求的能力。为此，一个全新的过滤器已被添加到 API
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+服务器聚合层的处理程序链中添加了一个新的过滤器。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:65]:
```suggestion
服务器聚合层的处理程序链中。
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+   it never proxies the request further.
+3. To set the network location of a kube-apiserver that peers will use to proxy requests,
+   the value passed in `--advertise-address` or (when `--advertise-address` is unspecified)
+   the `--bind-address` flag is used. For users with network configurations that would not
+   allow communication between peer kube-apiservers using the addresses specified in these flags,
+   there is an option to pass in the correct peer address as `--peer-advertise-ip` and
+   `--peer-advertise-port` flags that are introduced in this feature.
+-->
+1. 处理程序链中的新过滤器检查请求是否为 apiserver 无法解析的 API 组/版本/资源（使用现有的
+   [StorageVersion API](https://github.com/kubernetes/kubernetes/ blob/release-1.28/pkg/apis/apiserverinternal/types.go#L25-L37)）。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:89]:
```suggestion
   [StorageVersion API](https://github.com/kubernetes/kubernetes/blob/release-1.28/pkg/apis/apiserverinternal/types.go#L25-L37)）。
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+-->
+## 如何解决此问题？
+
+{{< figure src="/images/blog/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades/mvp-flow-diagram.svg" class="diagram-large" >}}
+
+<!--
+The new feature “Mixed Version Proxy” provides the kube-apiserver with the capability to proxy a request to a peer kube-apiserver which is aware of the requested resource and hence can serve the request. To do this, a new filter has been added to the handler chain in the API server's aggregation layer.
+-->
+"混合版本代理"新特性为 kube-apiserver 提供了将请求代理到对等 kube-apiserver
+以感知请求的资源并因此提供请求的能力。为此，一个全新的过滤器已被添加到 API
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:64]:
```suggestion
"混合版本代理"新特性为 kube-apiserver 提供了将请求代理到对等的、
能够感知所请求的资源并因此能够服务请求的 kube-apiserver。
为此，一个全新的过滤器已被添加到 API
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+   the value passed in `--advertise-address` or (when `--advertise-address` is unspecified)
+   the `--bind-address` flag is used. For users with network configurations that would not
+   allow communication between peer kube-apiservers using the addresses specified in these flags,
+   there is an option to pass in the correct peer address as `--peer-advertise-ip` and
+   `--peer-advertise-port` flags that are introduced in this feature.
+-->
+1. 处理程序链中的新过滤器检查请求是否为 apiserver 无法解析的 API 组/版本/资源（使用现有的
+   [StorageVersion API](https://github.com/kubernetes/kubernetes/blob/release-1.28/pkg/apis/apiserverinternal/types.go#L25-L37)）。
+   如果是，它会将请求代理到 ServerStorageVersion 对象中列出的 apiserver 之一。
+   如果识别的对等 apiserver 无法响应（由于网络连接、收到的请求与在 ServerStorageVersion
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:91]:
```suggestion
   如果所选的对等 apiserver 无法响应（由于网络连接、收到的请求与在 ServerStorageVersion
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+   对象中注册 apiserver-resource 信息的控制器之间的竞争等原因），则会出现 503（"Service Unavailable"）错误响应。
+2. 为了防止无限期代理请求，一旦确定原始请求无法处理该请求，就会在原始请求中添加一个（v1.28 新增）
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:93]:
```suggestion
2. 为了防止无限期地代理请求，一旦最初的 API 服务器确定无法处理该请求，就会在原始请求中添加一个
（v1.28 新增）
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+   HTTP 请求头 `X-Kubernetes-APIServer-Rerouted: true` API服务器。将其设置为 true 意味着原始
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:94]:
```suggestion
   HTTP 请求头 `X-Kubernetes-APIServer-Rerouted: true`。将其设置为 true 意味着原始
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+   API 服务器无法处理该请求，需要对其进行代理。如果目标对等 API 服务器看到此标头，它永远不会进一步代理该请求。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:95]:
```suggestion
   API 服务器无法处理该请求，需要对其进行代理。如果目标侧对等 API 服务器看到此标头，则不会对该请求做进一步的代理操作。
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+3. 要设置用于代理请求的 kube-apiserver 的对等点网络位置，请使用 `--advertise-address`
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:96]:
```suggestion
3. 要设置 kube-apiserver 的网络位置，以供对等服务器来代理请求，将使用 `--advertise-address`
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+   或（当未指定`--advertise-address`时）`--bind-address` 标志中传递的值。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:97]:
```suggestion
   或（当未指定`--advertise-address`时）`--bind-address` 标志所设置的值。
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+   如果网络配置中不允许用户使用这些标志中指定的地址在对等 kube-apiserver 之间进行通信，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:98]:
```suggestion
   如果网络配置中不允许用户在对等 kube-apiserver 之间使用这些标志中指定的地址进行通信，
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+   可以选择将正确的对等地址配置在此特性引入的 `--peer-advertise-ip` 和 `--peer-advertise-port`
+   参数中。
+
+<!--
+## How do I enable this feature?
+Following are the required steps to enable the feature:
+-->
+## 如何启用此特性？
+
+以下是启用该特性的步骤：
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:108]:
```suggestion
以下是启用此特性的步骤：
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 13]
```diff
@@ -0,0 +1,177 @@
...
+  destination kube-apiserver's serving certs using the flag `--peer-ca-file`
+  on the kube-apiservers. Note: this is a required flag for this feature to work.
+  There is no default value enabled for this flag.
+* Pass the correct ip and port of the local kube-apiserver that will be used by
+  peers to connect to this kube-apiserver while proxying a request.
+  Use the flags `--peer-advertise-ip` and `peer-advertise-port` to the kube-apiservers
+  upon startup. If unset, the value passed to either `--advertise-address` or `--bind-address`
+  is used. If those too, are unset, the host's default interface will be used.
+-->
+* 下载[最新的 Kubernetes 项目](/zh-cn/releases/download/)（版本 `v1.28.0` 或更高版本）
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:124]:
```suggestion
* 下载[Kubernetes 项目的最新版本](/zh-cn/releases/download/)（版本 `v1.28.0` 或更高版本）
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+* 在 kube-apiserver 上使用命令行标志 `--feature-gates=UnknownVersionInteroperabilityProxy=true`
+  打开特性门控
+* 传递源 kube-apiserver 将使用的 CA 证书包，使用 kube-apiserver 上 `--peer-ca-file`
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:127]:
```suggestion
* 使用 kube-apiserver 的 `--peer-ca-file` 参数为源 kube-apiserver 提供 CA 证书，
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+  的参数来验证目标 kube-apiserver 的服务证书。注意：这是此功能正常工作所必需的参数。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:128]:
```suggestion
  用以验证目标 kube-apiserver 的服务证书。注意：这是此功能正常工作所必需的参数。
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+  此参数没有默认值。
+* 传递本地 kube-apiserver 的正确 IP 和端口，在代理请求时，对等方将使用该 IP 和端口连接到此
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:130]:
```suggestion
* 为本地 kube-apiserver 设置正确的 IP 和端口，在代理请求时，对等方将使用该 IP 和端口连接到此
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 17]
```diff
@@ -0,0 +1,177 @@
...
+  kube-apiserver。使用为 kube-apiserver 设置的 `--peer-advertise-ip` 和
+  `--peer-advertise-port` 命令行参数。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:132]:
```suggestion
  kube-apiserver。在启动时，使用 `--peer-advertise-ip` 和
  `--peer-advertise-port` 命令行参数来配置 kube-apiserver。
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+  如果未设置，则使用 `--advertise-address` 或 `--bind-address` 命令行参数的值。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:133]:
```suggestion
  如果未设置这两个参数，则默认使用 `--advertise-address` 或 `--bind-address` 命令行参数的值。
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 19]
```diff
@@ -0,0 +1,177 @@
...
+
+目前，我们仅在确定时将资源请求代理到对等 kube-apiserver。
+接下来我们需要解决如何在这种情况下处理发现请求。
+目前我们计划在测试版中提供以下特性：
+
+<!--
+* Merged discovery across all kube-apiservers
+* Use an egress dialer for network connections made to peer kube-apiservers
+-->
+* 合并所有 kube-apiserver 的发现
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:150]:
```suggestion
* 合并所有 kube-apiserver 的发现数据
```

File: content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md [hunk 20]
```diff
@@ -0,0 +1,177 @@
...
+* 使用出口拨号器（egress dialer）与对等 kube-apiserver 进行网络连接
+
+<!--
+## How can I learn more?
+
+- Read the [Mixed Version Proxy documentation](/docs/concepts/architecture/mixed-version-proxy)
+- Read [KEP-4020: Unknown Version Interoperability Proxy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/4020-unknown-version-interoperability-proxy)
+-->
+## 怎样才能了解更多？
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md:159]:
```suggestion
## 如何进一步了解？
```

**Review by windsonsea — COMMENTED** [review]
Thanks.
/lgtm

**Pull request merged**